050-health-endpoint/proposal.md

```diff
@@ -0,0 +1,63 @@
+# Summary
+
+A public HTTP endpoint that gives basic information about the health of Concourse
+ cluster.
+
+
+# Motivation
+
+There are some scenarios in which a Concourse cluster is part of bigger setup for
+automation. In such scenario other parts of the system rely on Concourse not
+only for CI/CD, but for other type of automations as well. In order to have the
+whole process working the system also monitors the availability of its parts and
+ takes measures if some of its parts are not operational (for example executes
+some predefined steps - sends informing mails, triggers alerts, executes
+medication steps, etc). In such cases it is important for the system to be able
+to determine the state of every part of it. So it would be nice if the parts of
+this system have a common way to return their health/availability status.
+
+Currently there is no easy way for external (monitoring) system to understand
+if a Concourse cluster is live and operational. It would be nice if Concourse
+also gives information for its health so in (the rare) case it is not healthy
+the external system can react.
+
+
+# Proposal
+
+Concourse can expose a public HTTP endpoint called "health" endpoint that gives
```

Comment:
Would you mind elaborating a bit more on how you see the detection of "healthiness" would look like?

Also, do you think this endpoint should be public (in the sense of requiring no credentials at all) and have per-node information being displayed? I think a problem there is that it'd conflict with the current auth requirements that we have for someone reading `/api/v1/workers`.